docs/ref-manual/cvl/methods.md

```diff
@@ -46,44 +55,280 @@ method_summary   ::= "ALWAYS" "(" value ")"
                    | "HAVOC_ALL"
                    | "DISPATCHER" [ "(" ( "true" | "false" ) ")" ]
                    | "AUTO"
-                   | [ "with" "(" cvl_params ")" ] block
-                   | [ "with" "(" cvl_params ")" ] expression
+                   | [ "with" "(" "env" id ")" ] block
+                   | [ "with" "(" "env" id ")" ] expression
+                   | [ "with" "(" "env" id ")" ] id "(" [ id { "," id } ] ")"
 
-cvl_param ::= cvl_type [ id ]
+```
 
+```{todo}
+The `with(env id) block` and `with(env id) expression` productions come from
```

Comment:
The only valid expression summaries are applications of ghosts and cvl functions (I do not recall if we allow definitions but I suspect not)

Reply:
additionally, `with (env e)` does nothing and should not be used in summaries

Reply:
ok, I removed expression and blocks and with env, leaving only the function call syntax.  The section describing function summaries only lists CVL functions and ghosts as possiblities.